core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -34,23 +35,35 @@ object Entity {
     Default(body, length)
 
   // The type parameter aids type inference in Message constructors. This should be unnecessary when Messages are covariant.
-  def empty[F[_]]: Entity[F] = Empty
+  def empty[F[_]]: Entity[F] = _empty
+  private val _empty = new Strict(Chunk.empty)
+
+  def strict(chunk: Chunk[Byte]): Entity[Pure] = Strict(chunk)
 
   final case class Default[+F[_]](body: EntityBody[F], length: Option[Long]) extends Entity[F] {
     def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1] = that match {
       case d: Default[F1] => Default(body ++ d.body, (length, d.length).mapN(_ + _))
-      case Empty => this
+      case Strict(chunk) =>
+        if (chunk.isEmpty) this
+        else Default(body ++ Stream.chunk(chunk), length.map(_ + chunk.size))
     }
 
     def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G] = Default(body.translate(fk), length)
   }
 
-  case object Empty extends Entity[Pure] {
-    val body: EntityBody[Pure] = Stream.empty
+  final case class Strict(chunk: Chunk[Byte]) extends Entity[Pure] {
```

Comment:
Bikeshedding time :) I'm of the opinion we should use a scodec `ByteVector` here rather than a `Chunk`. Is there a reason you felt drawn to `Chunk`?

Reply:
Nope, just copying what @ChristopherDavenport did in the original PR. I'm totally open to `ByteVector` and see the appeal of not using an fs2 type here.

Reply:
Cool. My vote is for `ByteVector` then, curious to hear other's thoughts on this as well. My thinking is along the lines of:
1. `Chunk` is almost an fs2 internal thing here, as you point out
2. `ByteVector` is a lot more optimized for `Byte` than `Chunk[Byte]`
3. Precedent/defer to authority in https://github.com/http4s/http4s/pull/5321#discussion_r721641372

Reply:
Ok, the consensus on Discord is for `Chunk` primarily b/c `Chunk` can subsume `ByteVector` (but not the other way around), so this would let us avoid unnecessary copying.

Personally I still like `ByteVector` for the reasons above, but `Chunk[Byte]` is good too.